internal/EnumInterface.go

```
@@ -0,0 +1,6 @@
+package internal
+
+type enum[T int | string] interface {
```

golangci-lint (1.64.8)

type `enum` is unused

(unused)